### PR TITLE
Fix #134 - Stream filetrees as a tar file

### LINKED ISSUE #134: Add `tar` and `meta` resource routes for each resource
`/projects/{id}/files/tar` - Return a tar stream of all the files in the project
`/projects/{id}/files/meta` - Return the metadata for all the files in a project

`/files/{id}/tar` - Return a tar stream of the file
`/files/{id}/meta` - Return the metadata for the file

`/publishedProject/{id}/publishedFiles/tar` - Return a tar stream of the published files
`/publishedProject/{id}/publishedFiles/meta` - Return metadata for the published files

`/publishedFiles/{id}/tar` - Return a tar stream of the published file
`/publishedFiles/{id}/meta` - Return metadata for the published file

### PR DESCRIPTION
**Pull request by sedge**

### CODE REVIEW

File: api/classes/prerequisites.js [hunk 1]
```diff
@@ -21,20 +21,31 @@ var prerequisites = {};
  *
  * @return {Promise}
  */
-prerequisites.confirmRecordExists = function(model, mode, requestKey, databaseKey) {
-  databaseKey = databaseKey || requestKey;
+prerequisites.confirmRecordExists = function(model, config) {
```
Comment (gideonthomas) [inline api/classes/prerequisites.js]: much better!

File: api/modules/publishedFiles/routes/read.js [hunk 2]
```diff
@@ -41,4 +47,50 @@ module.exports = [{
...
+      prereqs.confirmRecordExists(Model, {
+        mode: 'param',
+        requestKey: 'published_id',
+        columns: ['id', 'published_id', 'file_id', 'path']
+      }),
+      prereqs.validateUser(),
+      prereqs.validateOwnership()
+    ],
+    handler: controller.getAllAsMeta.bind(controller),
+    description: 'Retrieve a collection of publishedFile objects that belong to a single project object, based on ' +
```
Comment (gideonthomas) [inline api/modules/publishedFiles/routes/read.js]: belong to a single *publishedProject* object

**Comment by gideonthomas** [comment]
r+ with nits and testing

**Comment by sedge** [comment]
I definitely need to add the tar header to those responses...

`"application/x-tar"`

**Comment by sedge** [comment]
Hmm, and stream error handling.

**Pull request merged**